dispatcher/brokers/pg_notify.py

```diff
@@ -129,6 +161,37 @@ def get_unlisten_query(self) -> psycopg.sql.SQL:
         """Stops listening on all channels for current session, see pg_notify docs"""
         return psycopg.sql.SQL("UNLISTEN *;")
 
+    async def initiate_self_check(self) -> None:
+        if self.self_check_status == BrokerSelfCheckStatus.IN_PROGRESS:
+            # another self-check message is in flight
+            raise RuntimeError(f'failed to initiate self check for broker {self.broker_id}')
+
+        try:
+            await self.apublish_message(self.self_check_channel, json.dumps({'self_check': True, 'sent': time.time(), 'task': f'lambda: "{self.broker_id}"'}))
+            self.self_check_status = BrokerSelfCheckStatus.IN_PROGRESS
+        except Exception as error:
+            raise RuntimeError(f'failed to send self check message: {error}')
+
+    def verify_self_check(self, message: dict[str, Any]) -> None:
+        """Verify a received self check message: check if it was sent from the same node and
+        is not outdated
+        """
+        now = time.time()
+        message_date = message['sent']
+        delta_seconds = now - message_date
```

Comment:
No problem sending the "time" in the message, but I would actually prefer to save the time on `self` when using it for comparison. This only requires that we only are running 1 self-check at a time, which I think is a safe assumption.

That would make it better to use `time.monotonic()` for, but that's not a meaningful global time stamp. If you are sending something in a message `time.time()` is better as a true epoch time, but `time.monotonic()` is less vulnerable to jumps due to time sync.